src/emulator/functionsEmulator.ts

```diff
@@ -514,6 +517,79 @@ You can probably fix this by running "npm install ${
     return loadTriggers();
   }
 
+  addRealtimeDatabaseTrigger(
+    projectId: string,
+    definition: EmulatedTriggerDefinition
+  ): Promise<any> {
+    const databasePort = EmulatorRegistry.getPort(Emulators.DATABASE);
+    if (!databasePort) {
+      EmulatorLogger.log(
+        "INFO",
+        `Ignoring trigger "${
+          definition.name
+        }" because the Realtime Database emulator is not running.`
+      );
+      return Promise.resolve();
+    }
+    if (definition.eventTrigger === undefined) {
+      EmulatorLogger.log(
+        "WARN",
+        `Event trigger "${definition.name}" has undefined "eventTrigger" member`
+      );
+      return Promise.reject();
+    }
+    /*
+     * The Realtime Database emulator expects the `path` field in its trigger
+     * definition to be relative to the database root.
+     */
+    const pathPattern = new RegExp("^projects/[^/]*/instances/[^/]*/refs(/.*)$");
```

Comment:
nit: Extract this out as a constant and move out.

Reply:
nit: We probably want `+` instead of `*` to make sure projects and instances id cannot be empty.